src/applications/PerfusionDerivatives.h

```diff
@@ -93,16 +97,72 @@ class PerfusionDerivatives
 
   template< class ImageType, class PerfusionImageType>
   typename ImageType::Pointer GetOneImageVolume(typename PerfusionImageType::Pointer perfImagePointerNifti, int index);
+
+  template< class ImageType, class PerfusionImageType >
+  std::vector<double> CalculateAveragePerfusionCurve(typename PerfusionImageType::Pointer perfImagePointerNifti);
+
+  template<class ImageType, class PerfusionImageType >
+  bool CheckPerfusionQuality(typename PerfusionImageType::Pointer perfImagePointerNifti, const std::string outputdirectory);
 };
 
 template< class ImageType, class PerfusionImageType >
-std::vector<typename ImageType::Pointer> PerfusionDerivatives::Run(std::string perfusionFile, bool rcbv, bool psr, bool ph, const double TE)
+std::vector<double> PerfusionDerivatives::CalculateAveragePerfusionCurve(typename PerfusionImageType::Pointer perfImagePointerNifti)
 {
-  std::vector<typename ImageType::Pointer> perfusionDerivatives;
-  perfusionDerivatives.push_back(NULL);
-  perfusionDerivatives.push_back(NULL);
-  perfusionDerivatives.push_back(NULL);
+  std::vector<double> AverageCurve;
+  ImageTypeFloat4D::RegionType region = perfImagePointerNifti->GetLargestPossibleRegion();
+  for (unsigned int volumes = 0; volumes < region.GetSize()[3]; volumes++)
```

Comment:
Instead of having 4 for-loops, perhaps we could use a simple itk Image iterator?

Reply:
replaced with iterator